source/material_model/utilities.cc

```diff
@@ -472,28 +472,98 @@ namespace aspect
           // Read data from disk and distribute among processes
           std::istringstream in(Utilities::read_and_distribute_file_content(filename, comm));
 
-          std::getline(in, temp); // eat first line
-          std::getline(in, temp); // eat next line
-          std::getline(in, temp); // eat next line
-          std::getline(in, temp); // eat next line
-
-          in >> min_temp;
-          std::getline(in, temp);
-          in >> delta_temp;
-          std::getline(in, temp);
-          in >> n_temperature;
-          std::getline(in, temp);
-          std::getline(in, temp);
-          in >> min_press;
-          min_press *= 1e5;  // conversion from [bar] to [Pa]
-          std::getline(in, temp);
-          in >> delta_press;
-          delta_press *= 1e5; // conversion from [bar] to [Pa]
-          std::getline(in, temp);
-          in >> n_pressure;
-          std::getline(in, temp);
-          std::getline(in, temp);
-          std::getline(in, temp);
+          // The following lines read in a PerpleX tab file in standard format
+          // The first 13 lines are a header in the format:
+          // |<perplex version>
+          // <table filename>
+          // <grid dim>
+          // <grid variable 1> (usually T(K) or P(bar))
+          // <min grid variable 1>
+          // <delta grid variable 1>
+          // <n steps grid variable 1>
+          // <grid variable 2> (usually T(K) or P(bar))
+          // <min grid variable 2>
+          // <delta grid variable 2>
+          // <n steps grid variable 2>
+          // Number of property columns in the table
+          // Column labels
+
+          // First line is the Perplex version number
+          std::getline(in, temp); // get next line, table file name
+
+          std::getline(in, temp); // get next line, dimension of table
+          unsigned int n_variables;
+          in >> n_variables;
+          AssertThrow (n_variables==2, ExcMessage("The PerpleX file must be two dimensional (P(bar)-T(K))."));
+
+          std::getline(in, temp); // get next line, either T(K) or P(bar)
+
+          for (unsigned int i=0; i<2; i++)
+            {
+              std::string natural_variable;
+              in >> natural_variable;
+
+              if (natural_variable.compare("T(K)") == 0)
```

Comment:
You've been hanging out with other languages too much that treat strings differently :-)
```suggestion
              if (natural_variable -- "T(K)")
```

Reply:
Guilty :)